catalog/core/catalog-core-encryption/src/main/java/org/codice/ddf/catalog/content/encryption/ContentEncryptorDecryptor.java

```diff
@@ -0,0 +1,280 @@
+import java.security.cert.CertificateException;
+import javax.crypto.Cipher;
+import javax.crypto.CipherInputStream;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ContentEncryptorDecryptor implements EncryptorDecryptor {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ContentEncryptorDecryptor.class);
+  private static final boolean CONTENT_ENCRYPTION_ENABLED =
+      SecurityConstants.getContentEncryptionEnabled();
+  private static final String CONTENT_ENCRYPTION_KEY_ALIAS_VALUE =
+      SecurityConstants.getContentEncryptionKeyAlias();
+  private static final String CONTENT_ENCRYPTION_KEY_PASSWORD_VALUE =
+      SecurityConstants.getContentEncryptionKeyPassword();
+  private static final String AES = "AES";
+  private static final String CIPHER_TYPE = "AES/GCM/NoPadding";
+  private static final String ENABLE_DEBUG_LOGGING_MSG =
+      " Enable debug logging for more information.";
+  private static final int KEY_LENGTH = 128;
+  private static final int KEY_BYTE_SIZE = KEY_LENGTH / 8;
+  private static final int IV_BYTE_SIZE = 12;
+  private static final int AUTH_TAG_LENGTH = 128;
+  private static final int AUTH_TAG_BYTE_SIZE = AUTH_TAG_LENGTH / 8;
+  public static final int EXTRA_BYTES = IV_BYTE_SIZE + AUTH_TAG_BYTE_SIZE;
+
+  private static SecureRandom secureRandom = new SecureRandom();
+  private static Cipher cipher;
+  private static KeyStore keyStore;
+  private static KeyStore trustStore;
+  private static SecretKey contentEncryptionKey;
+
+  private boolean isContentEncryptionEnabled;
+
+  public ContentEncryptorDecryptor() {
+    isContentEncryptionEnabled = SecurityConstants.getContentEncryptionEnabled();
+    if (!isContentEncryptionEnabled) {
+      return;
+    }
+    initKeystores();
+    initKey();
+    initCipher();
+  }
+
+  @VisibleForTesting
+  public ContentEncryptorDecryptor(SecretKey secretKey) {
+    isContentEncryptionEnabled = SecurityConstants.getContentEncryptionEnabled();
+    contentEncryptionKey = secretKey;
+    initCipher();
+  }
+
+  @Override
+  public InputStream encrypt(InputStream inputStream) throws ContentEncryptionException {
+    if (!isContentEncryptionEnabled) {
+      return inputStream;
+    }
+
+    readyCipherForEncryption();
+    return doEncryptionWithIV(cipher.getIV(), inputStream);
+  }
+
+  @Override
+  public InputStream decrypt(InputStream inputStream) throws ContentEncryptionException {
+    if (!isContentEncryptionEnabled) {
+      return inputStream;
+    }
+
+    if (!inputStream.markSupported()) {
+      LOGGER.debug("Wrapping InputStream in a BufferedInputStream to support marking.");
+      inputStream = new BufferedInputStream(inputStream);
+    }
+
+    byte[] iv = new byte[IV_BYTE_SIZE];
+    try {
+      inputStream.read(iv, 0, IV_BYTE_SIZE);
```

Comment:
❗️ Should check that the return value here is actually `IV_BYTE_SIZE`. If not, then it's an error condition.